### PR TITLE
feat: New Picker

### LINKED ISSUE #47063: DatePicker的format格式对于html的标签提示文本不生效
### Reproduction link

[`<DatePicker format={'DD/MM/YYYY'}/>`](https://ant.design/components/date-picker-cn)

官网demo在CodeSanbox中打开，format格式调整为"DD/MM/YYYY"
### Steps to reproduce

点击日期选择输入框，鼠标移入下方日期tag并停留一秒，观察弹出的提示性文本，仍然为YYYY-MM-DD格式
image

### What is expected?

提示性文本的格式与日期输入框的格式一致

### What is actually happening?

提示性文本的格式与日期输入框的格式不一致，对于国际化用户不友好

| Environment | Info |
| --- | --- |
| antd | 5.13.2 |
| React | 18.0.33 |
| System | win10 |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

Origin https://github.com/ant-design/ant-design/pull/46685

resolves #47063

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      【到 issue 中复制】     |
| 🇨🇳 Chinese |        【到 issue 中复制】   |

#### Chinese

* 🔥 重构 DatePicker 组件
  * ⚡️ 优化 `disabledDate` 检查逻辑，现在会提供 `info.type` 告知当前面板信息。
  * 🆕 DatePicker 新增 `multiple` 支持多选能力。
  * 🆕 新增配套 `pickerValue`、`defaultPickerValue` 属性用于受控管理展开面板所在日期。
    * `defaultPickerValue` 现在会在每次面板展开时都进行重置。
  * 🐞 修复输入框功能按键会唤起弹出框的问题，现在只有在交互按键、以及输入内容变化时唤起。
  * 🆕 新增 `preserveInvalidOnBlur` 属性用于无障碍时失去焦点需要保留输入内容的场景。
  * 🆕 `format` 支持 `align` 属性，允许通过掩码模式输入内容。
  * 🆕 支持 `required` 属性。
  * 🆕 RangePicker 下的 `id` 属性支持分别设置 `start` 和 `end` 输入框的 `id`。
  * 🆕 RangePicker 的 `onFocus` 和 `onBlur` 事件会额外提供一个 `info.range` 告知当前的焦点来自于哪个输入框。
  * 🆕 时间面板支持 `scrollOnChange` 设置滚动时间时自动选择对应的时间。
  * 🆕 添加 `needConfirm` 属性，允许设置需要确认、或者不确认的提交模式。
  * 🆕 添加 `components` 属性，允许自定义部分面板。
  * 🆕 面板中所有的日期相关信息都会允许通过 `locale` 进行配置。
  * 🆕 `format` 支持 `LT`、`LTS` 日期格式。
  * 🆕 新增 `minDate` 和 `maxDate` 用于设置面板切换范围。
  * 🆕 RangePicker 支持 `order` 属性。
  * 🆕 DatePicker 支持 `showWeek` 属性。
  * 🐞 通过输入框输入日期后失去焦点会提交变更，而不是丢失输入内容（即不再强制需要按下回车提交）。
  * 🐞 修复 `use12Hours` 下，`disabledTime` 获取的 `hour` 也会被裁剪成 0~12 的问题。
  * 🐞 修复某些交互下，禁用日期没有生效依然可以提交的问题。
  * 🛠 `allowClear` 触发事件从 `onMouseDown` 切换为 `onClick`。
  * 🛠 移除键盘对面板交互，它需要基于无障碍重新设计。
  * 🛠 废弃 `onKetDown` 的 `preventDefault` 参数，请直接通过 `event.preventDefault` 进行调用。
  * 💄 移除 RangePicker 的虚线样式，减少视觉干扰。
  * 💄 移除 RangePicker 在选择开始或者结束时间时的禁用范围，优化交互体验。

#### English

* 🔥 Refactor DatePicker component
  * ⚡️ Optimize `disabledDate` check logic, now it will provide `info.type` to inform the current panel information.
  * 🆕 DatePicker adds `multiple` to support multiple selection capabilities.
  * 🆕 Add matching `pickerValue` and `defaultPickerValue` attributes to control the date where the expanded panel is located.
    * `defaultPickerValue` will now be reset each time the panel is expanded.
  * 🐞 Fix the problem that the input box function key will trigger the pop-up box, now it will only be triggered when the interactive key and the input content change.
  * 🆕 Add the `preserveInvalidOnBlur` attribute to keep the input content when losing focus in the case of accessibility.
  * 🆕 `format` supports the `align` attribute, allowing input content through the mask mode.
  * 🆕 Support `required` attribute.
  * 🆕 The `id` attribute under RangePicker supports setting the `id` of the `start` and `end` input boxes separately.
  * 🆕 The `onFocus` and `onBlur` events of RangePicker will additionally provide an `info.range` to inform which input box the current focus comes from.
  * 🆕 The time panel supports `scrollOnChange` to automatically select the corresponding time when scrolling.
  * 🆕 Add the `needConfirm` attribute to allow setting the confirmation or non-confirmation submission mode.
  * 🆕 Add the `components` attribute to allow customizing some panels.
  * 🆕 All date-related information in the panel will allow configuration through `locale`.
  * 🆕 `format` supports `LT` and `LTS` date formats.
  * 🆕 Add `minDate` and `maxDate` to set the panel switching range.
  * 🆕 RangePicker supports `order` attribute.
  * 🆕 DatePicker supports `showWeek` attribute.
  * 🐞 Losing focus after entering the date through the input box will submit the change instead of losing the input content (that is, it is no longer necessary to press Enter to submit).
  * 🐞 Fix the problem that the `hour` obtained by `disabledTime` under `use12Hours` will also be clipped to 0~12.
  * 🐞 Fix the problem that the disabled date is not effective and can still be submitted under some interactions.
  * 🛠 `allowClear` trigger event from `onMouseDown` to `onClick`.
  * 🛠 Remove the keyboard interaction with the panel, it needs to be redesigned based on accessibility.
  * 🛠 Deprecate the `preventDefault` parameter of `onKetDown`, please call it directly through `event.preventDefault`.
  * 💄 Remove the dotted line style of RangePicker to reduce visual interference.
  * 💄 Remove the disabled range of RangePicker when selecting start or end time to optimize the interaction experience.
  

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by zombieJ** [comment]
- [ ] 图片 小时这一列的 disabled 颜色丢了。https://preview-46685-ant-design.surge.sh/components/date-picker-cn#components-date-picker-demo-disabled-date
- [ ] footer 高度变化
- [ ] maxDate、minDate、order、required 等属性貌似没有在 API 文档中出现。

**Comment by zombieJ** [comment]
Diff：
- [ ] `disabled` 弹窗变化，统一 SinglePicker & RangePicker 逻辑，禁用时不再允许打开面板，因而惰性计算状态。这个可以去保持一致。
- [x] compact 样式变化，使用了新的 token 计算方法，因而会变得“更紧凑”。
- [x] `mode` 强制设置为 `time` 时显示确认按钮。TimePicker 的确需要确认，符合预期。

**Review by li-jia-nan — APPROVED** [review]

File: components/calendar/__tests__/index.test.tsx [hunk 1]
```diff
@@ -16,7 +19,7 @@ import Select from '../../select';
 import Header, { type CalendarHeaderProps } from '../Header';
 
 const ref: {
-  calendarProps?: PickerPanelProps<unknown>;
+  calendarProps?: PickerPanelProps<any>;
```
Comment (crazyair) [inline components/calendar/__tests__/index.test.tsx:22]:
```suggestion
  calendarProps?: PickerPanelProps;
```

File: components/calendar/__tests__/index.test.tsx [hunk 2]
```diff
@@ -34,7 +37,7 @@ jest.mock('rc-picker', () => {
   const PickerPanelComponent = RcPicker.PickerPanel;
   return {
     ...RcPicker,
-    PickerPanel: (props: PickerPanelProps<unknown>) => {
+    PickerPanel: (props: PickerPanelProps<any>) => {
```
Comment (crazyair) [inline components/calendar/__tests__/index.test.tsx:40]:
```suggestion
    PickerPanel: (props: PickerPanelProps) => {
```

File: components/date-picker/__tests__/RangePicker.test.tsx [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
 import { resetWarned } from '../../_util/warning';
+import focusTest from '../../../tests/shared/focusTest';
+import { render, resetMockDate, screen, setMockDate, waitFor } from '../../../tests/utils';
 import enUS from '../locale/en_US';
-import { closeCircleByRole, expectCloseCircle, closePicker, openPicker, selectCell } from './utils';
+import { closeCircleByRole, closePicker, expectCloseCircle, openPicker, selectCell } from './utils';
 
 dayjs.extend(customParseFormat);
 
+type RangeValue<DateType extends object> =
```
Comment (crazyair) [inline components/date-picker/__tests__/RangePicker.test.tsx:16]: Picker 有 value 类型，要加个吗

File: components/date-picker/__tests__/RangePicker.test.tsx [hunk 4]
```diff
@@ -30,7 +34,7 @@ describe('RangePicker', () => {
   it('should not throw error when value is reset to `[]`', () => {
     const birthday = dayjs('2000-01-01', 'YYYY-MM-DD');
     const wrapper1 = render(<RangePicker value={[birthday, birthday]} open />);
-    const wrapper2 = render(<RangePicker value={[] as unknown as null} open />);
+    const wrapper2 = render(<RangePicker value={[] as unknown as RangeValue<dayjs.Dayjs>} open />);
```
Comment (crazyair) [inline components/date-picker/__tests__/RangePicker.test.tsx:37]:
```suggestion
    const wrapper2 = render(<RangePicker value={[]} open />);
```

File: components/date-picker/demo/disabled.tsx [hunk 5]
```diff
@@ -21,6 +21,11 @@ const App: React.FC = () => (
       defaultValue={[dayjs('2019-09-03', dateFormat), dayjs('2019-11-22', dateFormat)]}
       disabled={[false, true]}
     />
+    <DatePicker
+      defaultValue={dayjs('2019-09-03', dateFormat)}
+      minDate={dayjs('2019-06-01', dateFormat)}
```
Comment (crazyair) [inline components/date-picker/demo/disabled.tsx:26]: dateFormat 去掉，使用 plugins?

Comment (crazyair) [reply components/date-picker/demo/disabled.tsx:26]: customParseFormat 已经加了，dateFormat 可以去掉

File: components/date-picker/demo/suffix.tsx [hunk 6]
```diff
@@ -1,18 +1,12 @@
...
 const { RangePicker } = DatePicker;
 
-type DatePickerValue = Dayjs | null;
-type RangePickerValue = [Dayjs | null, Dayjs | null] | null;
-
-const onChange = (
-  date: DatePickerValue | RangePickerValue,
-  dateString: [string, string] | string,
-) => {
+const onChange = (date: Dayjs | (Dayjs | null)[] | null, dateString: string | string[]) => {
```
Comment (crazyair) [inline components/date-picker/demo/suffix.tsx:9]:
```suggestion
const onChange: PickerProps['onChange'] = (date, dateString) => {
```

File: components/date-picker/generatePicker/generateSinglePicker.tsx [hunk 7]
```diff
@@ -4,216 +4,199 @@ import CalendarOutlined from '@ant-design/icons/CalendarOutlined';
...
-    );
+              },
+            }}
+            allowClear={mergeAllowClear(allowClear, clearIcon, <CloseCircleFilled />)}
+          />
+        </NoCompactStyle>,
+      );
+    });
 
     if (displayName) {
```
Comment (crazyair) [inline components/date-picker/generatePicker/generateSinglePicker.tsx:193]: 为啥不是 process.env.NODE_ENV !== 'production'

File: components/date-picker/generatePicker/interface.ts [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+type InjectDefaultProps<Props> = Omit<Props, 'locale' | 'generateConfig' | 'hideHeader'> & {
+  locale?: PickerLocale;
+  size?: SizeType;
+  placement?: DataPickerPlacement;
+  /** @deprecated Use `variant` instead */
+  bordered?: boolean;
+  status?: InputStatus;
+  /**
+   * @since 5.13.0
```
Comment (crazyair) [inline components/date-picker/generatePicker/interface.ts:51]: 5.14.0

File: components/date-picker/index.ts [hunk 9]
```diff
@@ -43,4 +50,4 @@ const PureRangePanel = genPurePanel(DatePicker.RangePicker, 'picker', null, post
 (DatePicker as DatePickerType)._InternalRangePanelDoNotUseOrYouWillBeFired = PureRangePanel;
 (DatePicker as DatePickerType).generatePicker = generatePicker;
 
-export default DatePicker as DatePickerType;
+export default (DatePicker as DatePickerType);
```
Comment (crazyair) [inline components/date-picker/index.ts:53]: 这格式化一会加，一会去的

File: components/date-picker/__tests__/RangePicker.test.tsx (hunk 3 above)
Comment (zombieJ) [reply components/date-picker/__tests__/RangePicker.test.tsx:16]: test 里随便写写，外面的可以用新的 `GetProp` 拿到。

File: components/date-picker/demo/disabled.tsx (hunk 5 above)
Comment (zombieJ) [reply components/date-picker/demo/disabled.tsx:26]: 我感觉这个格式是不用特地加 dateFormat 的，不过这里为了尽量不要去动 demo 所以延续了写法。要改的话可以开个新的 PR 去做。

File: components/date-picker/index.ts (hunk 9 above)
Comment (zombieJ) [reply components/date-picker/index.ts:53]: 感觉加是对的，应该是以前 prettier 对 default 这个逻辑不太一样？

Comment (crazyair) [reply components/date-picker/index.ts:53]: 我这边是不加的

**Comment by Wxh16144** [comment]
点击面板外关闭面板的交互有 bug。
image

File: components/date-picker/demo/suffix.tsx (hunk 6 above)
Comment (zombieJ) [reply components/date-picker/demo/suffix.tsx:9]: 这个 onChange 是同时给 SinglePicker 和 RangePicker 使用的，不能使用 PickerProps

**Review by crazyair — APPROVED** [review]

**Comment by afc163** [comment]
cmponents 错别字

**Comment by crazyair** [comment]
> cmponents 错别字

@zombieJ https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker 插件安装下

**Comment by afc163** [comment]
footer 高度变化的问题修了么

**Comment by afc163** [comment]
图片

这几个高亮是符合预期的么？

**Comment by zombieJ** [comment]
#### footer 高度变化的问题修了么

Yes

#### 这几个高亮是符合预期的么？

Yes

File: docs/react/migration-v5.zh-CN.md [hunk 10]
```diff
@@ -259,6 +259,8 @@ export default {
 +   dayjs.locale('zh-cn');
 ```
 
+🚨 需要注意 day.js 通过插件系统拓展功能。如果你发现原本 moment.js 的功能在 day.js 中无法使用，请查阅 [day.js 官方文档](https://day.js.org/docs/en/plugin/plugin)。
```
Comment (yoyo837) [inline docs/react/migration-v5.zh-CN.md:262]: 怎么感觉在说我 :(

**Comment by li-jia-nan** [comment]
这个合掉之后那一堆 issue 就自动关了吗？

**Comment by yoyo837** [comment]
> 这个合掉之后那一堆 issue 就自动关了吗？

不会，还是需要写close, closes, fix, fixes 等

**Comment by AsuraKev** [comment]
Is this the Datepicker RFC? Does the custom panel support rendering datepicker in a modal for responsiveness?

**Comment by GitHubYuzai** [comment]
这个重构后 changeOnBlur被丢弃了啊

**Pull request merged**